osu.Game.Rulesets.Catch/Objects/CatchHitObject.cs

```diff
@@ -9,10 +9,11 @@
 using osu.Game.Rulesets.Objects;
 using osu.Game.Rulesets.Objects.Types;
 using osu.Game.Rulesets.Scoring;
+using osuTK;
 
 namespace osu.Game.Rulesets.Catch.Objects
 {
-    public abstract class CatchHitObject : HitObject, IHasXPosition, IHasComboInformation
+    public abstract class CatchHitObject : HitObject, IHasPosition, IHasComboInformation
```

Comment:
Making this implement `IHasYPosition`/`IHasPosition` is about the only qualm I have. Everything else in this diff is either conversion or legacy-related, which is fine, but I wouldn't want anything external to see that catch hitobjects have Y positions (because in any other contexts than legacy preservation that's just not true). Is this strictly required? Because I think I'd rather have some casting in places than this.

Reply:
The cast to the interface is used when it is encoded to the legacy beatmap format: 
https://github.com/ppy/osu/blob/7f432665e52774a5bef21cf930ff2ac1ac0527f6/osu.Game/Beatmaps/Formats/LegacyBeatmapEncoder.cs#L255